### ISSUE TITLE
[eslint-plugin-lit-a11y]: weird results

### ISSUE THREAD

**Issue by bennypowers** [open-wc/open-wc#1902]
I got some weird results for this template:

```ts
render(): TemplateResult {
  const launchpads = this.data?.launchpads ?? [];
  const landpads = this.data?.landpads ?? [];
  return html`
    <leaflet-map ?fit-to-markers="${!!this.data}">
      <leaflet-tilelayer
          url="https://{s}.basemaps.cartocdn.com/dark_all/{z}/{x}/{y}.png"
          max-zoom="19"
          .attribution="${attribution}"
      ></leaflet-tilelayer>
      ${launchpads.map(x => html`
        <leaflet-marker title="${x.name}"
            rise-on-hover
            longitude="${x.location.longitude}"
            latitude="${x.location.latitude}"
            data-id="${x.id}"
            @click="${this.onClickMarker}"
        ></leaflet-marker>
      `)}

      ${landpads.map(x => html`
        <leaflet-marker title="${x.name}"
            rise-on-hover
            longitude="${x.location.longitude}"
            latitude="${x.location.latitude}"
            data-id="${x.id}"
            @click="${this.onClickMarker}"
        ></leaflet-marker>
      `)}
    </leaflet-map>

    <p-card>
      <h2 slot="heading">${this.selected?.name ?? ''}</h2>
      <p>${this.selected?.details ?? ''}</p>
      <a href="${this.selected?.wikipedia ?? '#'}" target="_blank" rel="noopener noreferrer">Read More</a>
    </p-card>
  `;
}
```

> `<h2>` elements must have accessible content.

> The href attribute requires a valid value to be accessible. Provide a valid, navigable address as the href value. If you cannot provide a valid href, but still need the element to resemble a link, use a button and change it with appropriate styles. Learn more: https://github.com/evcohen/eslint-plugin-jsx-a11y/blob/master/docs/rules/anchor-is-valid.md

> Clickable non-interactive elements must have at least 1 keyboard event listener

> Clickable non-interactive elements must have at least 1 keyboard event listener

NB:

- The error on the h2 is not for the `<h2 slot="heading">`, rather it's on the this line:
```js
@click="${this.onClickMarker}"
```
- The error for the href is not on the `<p-card>`'s link, it's on this line:
```js
  ></leaflet-marker>
`)}
```
- WRT the click listeners, what about CEs that provide their own keyboard a11y? should be add an option to allowlist certain elements?
These might all be related to #1901

**Comment by thepassle** [comment]
Which version are you using? Some of these should be fixed, like the `<h2>` for example

**Comment by thepassle** [comment]
maybe custom-elements should just be excluded from the `click-events-have-key-events` entirely, what do you think?

**Comment by thepassle** [comment]
Can we chop this up in separate, more isolated issues?

As far as I can see there are three issues at play:
- Reporter reports wrong location, this is WIP upstream in eslint-plugin-lit
- custom-elements should be excluded from `click-events-have-key-events`, I think this is fixed in https://github.com/open-wc/open-wc/pull/1906
- optional chaining causes rule to fail which is fixed by https://github.com/open-wc/open-wc/pull/1907

**Comment by thepassle** [comment]
All these should be fixed now, by #1910  #1906 and #1907

**Comment by bennypowers** [comment]
confirmed fixed.
